ingestion/src/metadata/ingestion/source/metadata.py

```diff
@@ -334,6 +352,203 @@ def fetch_glossary_terms(self) -> GlossaryTerm:
                     break
                 after = glossary_term_entities.after
 
+    def fetch_locations(self) -> Location:
+        """fetch location method
+
+        Returns:
+            Location:
+        """
+        if self.service_connection.includeGlossaryTerms:
+            after = None
+            while True:
+                location_entities = self.metadata.list_entities(
+                    entity=Location,
+                    fields=["owner", "followers", "tags"],
+                    after=after,
+                    limit=self.service_connection.limitRecords,
+                )
+                for location in location_entities.entities:
+                    self.status.scanned_team(location.name)
+                    yield location
+                if location_entities.after is None:
+                    break
+                after = location_entities.after
+
+    def fetch_charts(self) -> Chart:
+        """fetch charts method
+
+        Returns:
+            Chart:
+        """
+        after = None
+        while True:
+            chart_entities = self.metadata.list_entities(
```

Comment:
This logic seems that could be reused with a function accepting `entity: Type[T], fields: List[str]` 🤔